### ISSUE TITLE
Binding does not propagate to nested item templates

### ISSUE THREAD

**Issue by amin-nas** [EnsembleUI/ensemble#403]
```yaml
View:
  onLoad:
    invokeAPI:
      name: getUser
      inputs:
        q: jack

  body:
    Column:
      children:
        - Button:
            label: Call api
            onTap:
              invokeAPI:
                name: getUser
                inputs:
                  q: john
        - Column:
            item-template:
              data: ${getUser.body.users}
              name: u
              template: 
                Column:
                  children:
                    - Text:
                        text: ${u.firstName}
                    - Column:
                        item-template:
                          data: ${u.items}
                          name: i
                          template:
                            Text:
                              text: ${i}




API:
  getUser:
    inputs:
      - q
    uri: https://dummyjson.com/users/search?q=${q}
    method: GET
    onResponse: |
      //@code
      var rand1 = Math.floor(Math.random() * 10);
      var rand2 = Math.floor(Math.random() * 10);
      console.log("New random numbers are");
      console.log(rand1);
      console.log(rand2);

      response.body.users.map(function(u) {
        u.items = [];
        u.items.push(rand1);
        u.items.push(rand2);
      });
```

**Comment by vusters** [comment]
This used to work. Sending this to you @vinothvino42

**Comment by vinothvino42** [comment]
It's closed now with the PR #407
